### PR TITLE
BOSA21Q1-894 Add auth handler for participants impersonations

### PR DESCRIPTION
**Pull request by sergei-krylov**

### CODE REVIEW

File: app/services/dummy_authorization_handler.rb [hunk 1]
```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+# An example authorization handler used so that users can be verified against
+# third party systems.
+#
+# You should probably rename this class and file to match your needs.
```
Comment (simon-lejeune) [inline app/services/dummy_authorization_handler.rb:6]: shouldn't we?

File: app/services/dummy_authorization_handler.rb [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    errors.add(:document_number, :invalid) unless document_number =~ /\A[a-zA-Z0-9]+\z/
+  end
+
+  def valid_scope_id
+    errors.add(:scope_id, :invalid) if scope_id && !scope
+  end
+
+  # If you need custom authorization logic, you can implement your own action
+  # authorizer. In this case, it allows to set a list of valid postal codes for
+  # an authorization.
```
Comment (simon-lejeune) [inline app/services/dummy_authorization_handler.rb:87]: this is the decidim example copy pasted right?

Comment (sergei-krylov) [reply app/services/dummy_authorization_handler.rb:87]: Yes, it is an example from decidim.

**Comment by sergei-krylov** [comment]
In general, I'm not sure about renaming it to something else just because Bosa don't have anything specific for that action (impersonation) in terms of fields, rules, etc. Just to make it clearer in the code - yes we can.
Maybe `ParticipantsImpersonationsHandler`? According to - https://docs.decidim.org/en/admin/participants/impersonations.html

**Review by simon-lejeune — APPROVED** [review]

**Review by simon-lejeune — APPROVED** [review]

**Comment by sergei-krylov** [comment]
According to discussion above - the auth handler is renamed to `ParticipantImpersonationHandler`.
Because of renaming it is also needed to have the `form.html.erb` to properly display the attributes - scope and birthday.
Another thing I forgot in my initial change is that this handler shouldn't be available in user settings (`/authorizations` page for logged in user), for that there are:
- code piece to handle custom extends , in [config/application.rb](https://github.com/belighted/bosa-cities-new/pull/8/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9)
- extend for authorizations controller to not include the specific handled in the list, in [.../decidim-verifications/app/controllers/decidim/verifications/authorizations_controller.rb](https://github.com/belighted/bosa-cities-new/pull/8/files#diff-9c11706d082b2e18f20c0c710839e1de5c1a19aec72ad6a7d7f9d8fa0b214374)

**Pull request merged**